src/SiteConfig/ConfigBuilder.php

```diff
@@ -327,6 +327,15 @@ public function mergeConfig(SiteConfig $currentConfig, SiteConfig $newConfig)
         //      find_string: <other-img
         //      replace_string: <img
         // To fix that issue, we combine find & replace as key & value in one array, we merge them and then rebuild find & replace string in the current config
+
+        // merge http_header array from currentConfig into newConfig
+        // because final values override former values in case of named keys
+        $currentConfig->http_header = array_merge($newConfig->http_header, $currentConfig->http_header);
+
+        if (\count($currentConfig->find_string) !== \count($currentConfig->replace_string)) {
+            return $currentConfig;
+        }
```

Comment:
This looks weird to me. Sohuld not we rather return `$newConfig`? And should not we check the variables from `$newConfig`? I would assume the `$currentConfig` has been already validated upon construction.

Reply:
You're right, I'll move that check in `parseLines` instead